### ISSUE TITLE
Class not found when unmarshalling ...CropImageOptions

### ISSUE THREAD

**Issue by vasskob** [ArthurHub/Android-Image-Cropper#329]
Have an error only on Samsung Galaxy A5, 7.0.
Class not found when unmarshalling: com.theartofdev.edmodo.cropper.CropImageOptions
![screenshot from 2017-07-05 10-04-42](https://user-images.githubusercontent.com/7678264/27854074-02f9fdb6-616e-11e7-8de3-51e8ebab3e97.png)

and then JNI 
![screenshot from 2017-07-05 10-38-21](https://user-images.githubusercontent.com/7678264/27854045-efaa5a26-616d-11e7-9fa4-58f65f936326.png)
![screenshot from 2017-07-05 10-05-44](https://user-images.githubusercontent.com/7678264/27854057-f8230e1e-616d-11e7-8279-886c63a2c6bb.png)
No JNI issue on Nexus 5, 7.1.2, Samsung Grand 5.1.1, Samsung S7 edge 7.0, but Class not found CropImageOptions still present.

**Comment by Vantablack** [comment]
This happens to me too! I'm testing it on `Samsung I9300 Galaxy S III 4.3`.

I'm following the guide [Using Activity](https://github.com/ArthurHub/Android-Image-Cropper#using-activity):

```java
...

// start cropping activity for pre-acquired image saved on the device
CropImage.activity(imageUri)
 .start(this);

...

```

InvokerActivity:

```java
CropImage.activity(selectedImageUri)
                    .setAspectRatio(1, 1)
                    .setFixAspectRatio(true)
                    .start(this);
```

AndroidManifest.xml:

```xml
<application>
    ... 
    <activity
            android:name="com.theartofdev.edmodo.cropper.CropImageActivity"
            android:screenOrientation="portrait"
            android:theme="@style/Base.Theme.AppCompat" />
    ...
</application>
```

My stacktrace:
```
07-06 21:51:59.730 2353-5391/? E/Parcel: Class not found when unmarshalling: com.theartofdev.edmodo.cropper.CropImageOptions
                                         java.lang.ClassNotFoundException: com.theartofdev.edmodo.cropper.CropImageOptions
                                             at java.lang.Class.classForName(Native Method)
                                             at java.lang.Class.forName(Class.java:204)
                                             at java.lang.Class.forName(Class.java:169)
                                             at android.os.Parcel.readParcelableCreator(Parcel.java:2091)
                                             at android.os.Parcel.readParcelable(Parcel.java:2055)
                                             at android.os.Parcel.readValue(Parcel.java:1971)
                                             at android.os.Parcel.readMapInternal(Parcel.java:2255)
                                             at android.os.Bundle.unparcel(Bundle.java:223)
                                             at android.os.Bundle.getString(Bundle.java:1082)
                                             at android.content.Intent.getStringExtra(Intent.java:4961)
                                             at com.android.server.am.ActivityStack.startActivityLocked(ActivityStack.java:3761)
                                             at com.android.server.am.ActivityStack.startActivityMayWait(ActivityStack.java:4977)
                                             at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:3173)
                                             at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3129)
                                             at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:157)
                                             at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2125)
                                             at android.os.Binder.execTransact(Binder.java:388)
                                             at dalvik.system.NativeStart.run(Native Method)
                                          Caused by: java.lang.NoClassDefFoundError: com/theartofdev/edmodo/cropper/CropImageOptions
                                             at java.lang.Class.classForName(Native Method) 
                                             at java.lang.Class.forName(Class.java:204) 
                                             at java.lang.Class.forName(Class.java:169) 
                                             at android.os.Parcel.readParcelableCreator(Parcel.java:2091) 
                                             at android.os.Parcel.readParcelable(Parcel.java:2055) 
                                             at android.os.Parcel.readValue(Parcel.java:1971) 
                                             at android.os.Parcel.readMapInternal(Parcel.java:2255) 
                                             at android.os.Bundle.unparcel(Bundle.java:223) 
                                             at android.os.Bundle.getString(Bundle.java:1082) 
                                             at android.content.Intent.getStringExtra(Intent.java:4961) 
                                             at com.android.server.am.ActivityStack.startActivityLocked(ActivityStack.java:3761) 
                                             at com.android.server.am.ActivityStack.startActivityMayWait(ActivityStack.java:4977) 
                                             at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:3173) 
                                             at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3129) 
                                             at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:157) 
                                             at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2125) 
                                             at android.os.Binder.execTransact(Binder.java:388) 
                                             at dalvik.system.NativeStart.run(Native Method) 
                                          Caused by: java.lang.ClassNotFoundException: Didn't find class "com.theartofdev.edmodo.cropper.CropImageOptions" on path: .
                                             at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:64)
                                             at java.lang.ClassLoader.loadClass(ClassLoader.java:501)
                                             at java.lang.ClassLoader.loadClass(ClassLoader.java:461)
                                             at java.lang.Class.classForName(Native Method) 
                                             at java.lang.Class.forName(Class.java:204) 
                                             at java.lang.Class.forName(Class.java:169) 
                                             at android.os.Parcel.readParcelableCreator(Parcel.java:2091) 
                                             at android.os.Parcel.readParcelable(Parcel.java:2055) 
                                             at android.os.Parcel.readValue(Parcel.java:1971) 
                                             at android.os.Parcel.readMapInternal(Parcel.java:2255) 
                                             at android.os.Bundle.unparcel(Bundle.java:223) 
                                             at android.os.Bundle.getString(Bundle.java:1082) 
                                             at android.content.Intent.getStringExtra(Intent.java:4961) 
                                             at com.android.server.am.ActivityStack.startActivityLocked(ActivityStack.java:3761) 
                                             at com.android.server.am.ActivityStack.startActivityMayWait(ActivityStack.java:4977) 
                                             at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:3173) 
                                             at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3129) 
                                             at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:157) 
                                             at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2125) 
                                             at android.os.Binder.execTransact(Binder.java:388) 
                                             at dalvik.system.NativeStart.run(Native Method) 
```

**Comment by ArthurHub** [comment]
Are you using pro guard?

**Comment by Vantablack** [comment]
@ArthurHub As far as I know, nope. 

Here's my `build.gradle` and so far I'm using build type `debug`

```gradle
android {
    compileSdkVersion 25
    buildToolsVersion '25.0.3'

    defaultConfig {
        applicationId "com.reachfieldit.rsms.mobile"
        minSdkVersion 16
        targetSdkVersion 18
        versionCode 1
        versionName "1.0"
    }

    buildTypes {
        debug {
            applicationIdSuffix ".debug"
            versionNameSuffix "-DEBUG"
        }
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }

    productFlavors {
        development {
            applicationIdSuffix ".development"
            versionNameSuffix "-DEV"
        }

        production {
            applicationIdSuffix ".production"
            versionNameSuffix "-PROD"
        }
    }
}
```

**Comment by ArthurHub** [comment]
Hmm...
Is it possible that one of the other libraries you use also references this one?
found this: http://www.gitshah.com/2011/04/fixing-javalangclassnotfoundexception.html

**Comment by Vantablack** [comment]
@ArthurHub 

I have Gson in my dependencies but in that activity I did not use/initialize Gson

```gradle
dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    compile 'com.android.support:appcompat-v7:25.3.1'
    compile 'com.android.support:design:25.3.1'
    compile 'com.google.android.gms:play-services-location:10.0.1'
    compile 'com.google.code.gson:gson:2.8.0'
    compile 'com.squareup.okhttp3:okhttp:3.8.0'
    compile 'com.facebook.stetho:stetho:1.4.2'
    compile 'com.facebook.stetho:stetho-okhttp3:1.4.2'
    compile 'com.android.support.constraint:constraint-layout:1.0.2'
    compile 'com.theartofdev.edmodo:android-image-cropper:2.4.6'
    compile 'com.squareup.okhttp3:mockwebserver:3.8.1'
    testCompile 'junit:junit:4.12'
}
```

**Comment by ArthurHub** [comment]
Samsung...
can you try the solution described here: https://stackoverflow.com/questions/28589509/android-e-parcel-class-not-found-when-unmarshalling-only-on-samsung-tab3

**Comment by Vantablack** [comment]
@ArthurHub Sorry to trouble you, but I'm not sure how to set it up as I'm not passing a Parcel around. For more context, I'm invoking `CropImage#activity()` in `onActivityResult` which is after I take a photo.

```java
@Override
protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    // Handle initial image capture
    if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK) {
        Uri selectedImageUri = Uri.fromFile(new File(mCurrentPhotoPath));

        CropImage.activity(selectedImageUri)
                .setAspectRatio(1, 1)
                .setFixAspectRatio(true)
                .start(this);
    }
    // Handle crop image result
    if (requestCode == CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE && resultCode == RESULT_OK) {

        CropImage.ActivityResult result = CropImage.getActivityResult(data);

        Uri croppedImageUri = result.getUri();
        try {
            Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), croppedImageUri);
            mImageView.setImageBitmap(bitmap);
        } catch (IOException e) {
            Log.i(TAG, "onActivityResult: imageUri is not valid");
            e.printStackTrace();
        }

        // Revoke URI permission
        this.revokeUriPermission(Uri.fromFile(new File(mCurrentPhotoPath)),
                Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
    }
}
```

**Comment by ArthurHub** [comment]
Yea... I need your help here. Because I can't test it I need you to change the code in the library.
You should be able to reproduce the issue in the "quick-start" sample app that is part of this project.
And then change the code to `CropImage.getIntent()` and `CropImageActivity.onCreate()` methods.

**Comment by ArthurHub** [comment]
Fixed in 2.4.7
